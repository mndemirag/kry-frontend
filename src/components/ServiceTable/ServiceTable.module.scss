@import "../../utils";

table {
  display: block;
  word-wrap: break-word;
  text-align: left;

  @media (min-width: $screen-md-min) {
    display: table;
    width: 100%;
  }

  thead,
  tbody,
  th,
  td,
  tr {
    display: block;
  }

  // Revert back to default table styling in Desktop
  @media (min-width: $screen-md-min) {
    tr {
      display: table-row;
    }
    th,
    td {
      display: table-cell;
    }
    thead {
      display: table-header-group;
    }
    tbody {
      display: table-row-group;
    }
  }

  thead tr {
    @media (max-width: $screen-md-min) {
      @include visuallyHidden;
    }
  }

  tr {
    border-bottom: 1px solid lightgray;
    padding-bottom: 12px;
    padding-top: 12px;

    &:last-child {
      border-bottom: none;
    }
  }

  td {
    /* Make it a row */
    position: relative;
    padding-left: 50%;
    padding-bottom: 8px;

    @media (min-width: $screen-md-min) {
      padding: 0px;

      &:before {
        display: none;
      }
    }

    &:before {
      /* Table header */
      position: absolute;
      top: 6px;
      left: 6px;
      padding-right: 10px;
      white-space: nowrap;
      font-weight: bold;
    }

    /* Label the data */
    &:nth-of-type(1):before {
      content: "Name";
    }
    &:nth-of-type(2):before {
      content: "Status";
    }
    &:nth-of-type(3):before {
      content: "Url";
    }
    &:nth-of-type(4):before {
      content: "Actions";
    }
  }

  .nameCol {
    width: 20%;
  }

  .statusCol {
    width: 15%;
  }
}

.tableTitle {
  width: 50%;
  padding-bottom: 10px;
  margin: 0 0 10px;
  display: flex;
  justify-content: space-between;
  align-items: center;
  text-transform: uppercase;
}
