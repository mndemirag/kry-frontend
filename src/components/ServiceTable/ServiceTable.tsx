import { ServiceTableProps } from "../../types";
import { DeleteIcon } from "../Icons/DeleteIcon";

import styles from "./ServiceTable.module.scss";

export const ServiceTable: React.FC<ServiceTableProps> = ({
  data,
  handleDelete,
}) => {
  return (
    <>
      <div className={styles.tableTitle} role="heading" aria-level={1}>
        <h2>Service Table</h2>
      </div>
      <table>
        <thead>
          <tr>
            <th className={styles.nameCol}>Name</th>
            <th className={styles.statusCol}>Status</th>
            <th>Url</th>
            <th>Actions</th>
          </tr>
        </thead>
        <tbody>
          {data.map((item, i) => (
            <tr key={i}>
              <td>{item.name}</td>
              <td>{item.status.status}</td>
              <td>{item.status.url}</td>
              <td>
                <div
                  className={styles.iconWrapper}
                  onClick={() => handleDelete(item.name)}
                >
                  <DeleteIcon />
                </div>
              </td>
            </tr>
          ))}
        </tbody>
      </table>
    </>
  );
};
