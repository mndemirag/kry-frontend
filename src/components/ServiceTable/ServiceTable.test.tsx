import { render } from "@testing-library/react";

import { ServiceTable } from "./ServiceTable";

const mockDelete = jest.fn();

const mockProps = {
  data: [
    {
      name: "melodi",
      status: { name: "melodi", url: "https://melodi.com", status: "OK" },
    },
  ],
  handleDelete: mockDelete,
};

describe("ServiceTable", () => {
  it("renders properly", () => {
    const { container } = render(
      <ServiceTable data={mockProps.data} handleDelete={mockDelete} />
    );
    expect(container).toMatchSnapshot();
  });
});
