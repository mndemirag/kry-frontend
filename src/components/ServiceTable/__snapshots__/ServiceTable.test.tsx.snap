// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServiceTable renders properly 1`] = `
<div>
  <div
    aria-level="1"
    class="tableTitle"
    role="heading"
  >
    <h2>
      Service Table
    </h2>
  </div>
  <table>
    <thead>
      <tr>
        <th
          class="nameCol"
        >
          Name
        </th>
        <th
          class="statusCol"
        >
          Status
        </th>
        <th>
          Url
        </th>
        <th>
          Actions
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          melodi
        </td>
        <td>
          OK
        </td>
        <td>
          https://melodi.com
        </td>
        <td>
          <div
            class="iconWrapper"
          >
            <div
              id="deleteIcon"
            >
              <svg
                aria-hidden="true"
                height="24px"
                viewBox="0 0 48 48"
                width="24px"
                xmlns="http://www.w3.org/2000/svg"
              >
                <polyline
                  fill="#f55376"
                  points="12.229,42.048 5.952,35.77 35.771,5.952 42.048,12.23 12.229,42.048"
                />
                <path
                  d="M30.278,24L24,30.278l11.771,11.77l6.277-6.278L30.278,24 M12.229,5.952L5.952,12.23L17.722,24 L24,17.722L12.229,5.952"
                  fill="#f55376"
                />
                <polyline
                  fill="#eb0000"
                  points="24,17.722 17.722,24 24,30.278 30.278,24 24,17.722"
                />
              </svg>
              <span
                class="visuallyHidden"
              >
                Delete the service
              </span>
            </div>
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
